### ISSUE TITLE
Follow secure coding practices 

### ISSUE THREAD

**Issue by bayotop** [stripe/stripe-dotnet#993]
There are two minor issues in the `StripeEventUtility.computeSignature()` method (https://github.com/stripe/stripe-dotnet/blob/ff283e7c50914e2847ecf9fd70b896950fe87ded/src/Stripe.net/Services/Events/StripeEventUtility.cs):

1. The `HMACSHA256` object should be explicitly disposed. It is important to dispose cryptographic objects ASAP given that these objects wipe internal buffers on disposal, which prevents sensitive data from sitting in memory indefinitely. Note that `HMACSHA256` has no finalizer so the GC may not ever trigger a disposal.
2. The second issue is the default behaviour of `Encoding.UTF8.GetBytes(string)`. Unicode codepoints without a valid UTF-8 representation will by default fallback and replace the failed mapping with "?" (See https://msdn.microsoft.com/en-us/library/system.text.encoding.utf8(v=vs.110).aspx). The problem can be seen in the following example: 
```csharp
string secret = "\ud801";
string secret2 = "\ud802";

if (!secret.Equals(secret2))
{
    Console.Out.WriteLine("The strings are not equal.");
}

computeSignature(secret, "timestamp", "payload");  // returns 1eba7aac5f10ee7aa985722256aa4125e8e59fe256386c8ab10295063d986e57
computeSignature(secret2, "timestamp", "payload"); // returns 1eba7aac5f10ee7aa985722256aa4125e8e59fe256386c8ab10295063d986e57
```

The example shows that the method calculated the same signature for two different keys. The reason is that both secrets became "?" because the `GetBytes()` method did silently fallback.

Both issues can be fixed quite easily. In the first case simply wrapping the `HMACSHA256` object into a using block will do the trick. In the latter case you should use a `UTF8Encoding` instance that throws an exception on failed mappings to prevent the situation that identical signatures are generated for different keys

**Comment by brandur-stripe** [comment]
Thanks again for sticking through this @bayotop. We've released your fix as part of v11.6.0.

**Comment by ob-stripe** [comment]
Closed by #1035.
